libparsec/crates/client/tests/unit/certificates_ops/store.rs

```diff
@@ -0,0 +1,190 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
+
+use std::sync::Arc;
+
+use libparsec_platform_storage::certificates::GetCertificateError;
+use libparsec_tests_fixtures::prelude::*;
+use libparsec_types::prelude::*;
+
+use super::utils::certificates_store_factory;
+
+#[parsec_test(testbed = "minimal", with_server)]
```

Comment:
It's better to avoid relying on the testbed server as much as possible (given it makes the test much slower)

It should be easy to do that for all tests involving the certificate ops:
- only the poll function communicates with the server
- the poll function is triggered only by the validate functions

so:
- you can test the `add_certificates_batch` function with absolutely no server
- for testing the validation function, you should use the testbed customize function to specify the certificate ops storage should be already populated for device alice@dev1 (see what is done for the minimal_client_sync template)
- for testing the poll command, it is better anyway to use the testbed request mock (again for speed, but also to ensure what command is needed to do this action)

Reply:
Outdated